internal/dag/policy.go

```diff
@@ -57,6 +57,9 @@ const (
 	LoadBalancerPolicyRequestHash = "RequestHash"
 )
 
+// match "%REQ(<X-Foo-Bar>)%"
+var hostRewriteHeaderRegex = regexp.MustCompile(`%REQ\((\S+)\)%`)
```

Comment:
to be more exact here we can use `[-A-Za-z0-9]+` as the regex to capture the header name

see: https://github.com/kubernetes/apimachinery/blob/a017454658b6eb672eb2d7b5eedad67f5f67665c/pkg/util/validation/validation.go#L395-L407

Reply:
should ensure we're not accidentally capturing invalid characters/header names and sending to envoy

Reply:
yup makes sense. added in https://github.com/projectcontour/contour/pull/5678/commits/846ab1c682fd1a5e054a214061236d1e803355bf